pyro/infer/enum.py

```diff
@@ -3,18 +3,28 @@
 import math
 
 import torch
+from six.moves.queue import LifoQueue
 from torch.autograd import Variable
 
 from pyro import poutine
+from pyro.distributions.util import sum_rightmost
 from pyro.poutine.trace import Trace
-from six.moves.queue import LifoQueue
 
 
-def site_is_discrete(name, site):
-    return getattr(site["fn"], "enumerable", False)
+def _iter_discrete_filter(name, msg):
```

Comment:
Can we remove the name from the function args?

Reply:
I can't remove this arg without changes to `Trace.compute_batch_log_pdf()`. Let's clean this up later when we refactor the `Trace.compute_()` methods.